File: Python/BFS/LevelOrderTraversal.py
# @author
# Aakash Verma

# Output:
# [1]
# [2, 3]
# [4, 5, 6, 7] 

from collections import *

class Node:
    def __init__(self, data):
        self.data = data
        self.right = self.left = None

class LevelOrderTraversal:
    
    def __init__(self): 
        self.root = None

    def traverse(self, root):
        bfs = []
        if root is None:
            return bfs
        queue = deque([])
        queue.append(self.root)
        while len(queue) != 0:
            level_size = len(queue)
            current_level = []
            for i in range(level_size):
                current = queue.popleft()
                current_level.append(current.data)
                if current.left is not None:
                    queue.append(current.left)
                if current.right is not None:
                    queue.append(current.right)
            bfs.append(current_level)
        return bfs
    
if __name__ == '__main__':
    tree = LevelOrderTraversal()
    tree.root = Node(1)
    tree.root.left = Node(2)
    tree.root.right = Node(3)
    tree.root.left.left = Node(4)
    tree.root.left.right = Node(5)
    tree.root.right.left = Node(6)
    tree.root.right.right = Node(7)
    ans = tree.traverse(tree.root)
    for level in ans:
        print(level)

File: Java/BFS/LevelOrderTraversal.java
/*
*   @author
*   Aakash Verma
*
*   Output:
*   [1]
*   [2, 3]
*   [4, 5, 6, 7]
*
*/

import java.util.*;

class Node {
    int data;
    Node left, right;
    public Node(int key) {
        data = key;
        left = right = null;
    }
}


class LevelOrderTraversal {

    static Node root;
    LevelOrderTraversal() {
        root = null;
    }

    public static List<List<Integer>> traverse(Node root) {
        List<List<Integer>> bfs = new ArrayList<>();
        if(root == null) return bfs;
        Queue<Node> queue = new LinkedList<>();
        queue.offer(root);
        while(!queue.isEmpty()) {
            int levelSize = queue.size();
            List<Integer> currentLevel = new ArrayList<>();
            for(int i = 0; i < levelSize; i++) {
                Node current = queue.poll();
                currentLevel.add(current.data);
                if(current.left != null) 
                    queue.offer(current.left);
                if(current.right != null)
                    queue.offer(current.right);
            }
            bfs.add(currentLevel);
        }
        return bfs;
    }

    public static void main (String[] args) {
        root = new Node(1);
        root.left = new Node(2);
        root.right = new Node(3);
        root.left.left = new Node(4);
        root.left.right = new Node(5);
        root.right.left = new Node(6);
        root.right.right = new Node(7);
        List<List<Integer>> ans = traverse(root);
        for(List<Integer> level : ans) {
            System.out.println(level.toString());
        }
    }
}